src/main/java/org/spongepowered/common/mixin/core/entity/weather/MixinEntityLightningBolt.java

```diff
@@ -57,12 +71,31 @@ public ExpirableData getExpiringData() {
     @Inject(method = "onUpdate()V", at = {@At(value = "NEW", args = "class=net.minecraft.util.BlockPos"),
             @At(value = "NEW", args = "class=net.minecraft.util.AxisAlignedBB")}, cancellable = true
         )
-    public void onOnUpdate(CallbackInfo ci) {
+    public void onDoDamage(CallbackInfo ci) {
         if (this.effect) {
             ci.cancel();
         }
     }
 
+    @Inject(method = "onUpdate()V", at = @At(value = "INVOKE",
+            target = "Lnet/minecraft/world/World;setBlockState(Lnet/minecraft/util/BlockPos;Lnet/minecraft/block/state/IBlockState;)Z"))
+    public void onStrikeBlock(World world, BlockPos pos, IBlockState blockState) {
```

Comment:
Need to fully qualify `World` (`net.minecraft.world.World`) because the imported world is the sponge API world.